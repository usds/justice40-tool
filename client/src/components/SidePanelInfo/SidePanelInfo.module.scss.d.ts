declare namespace MapIntroductionModuleScssNamespace {
    export interface IMapIntroductionModuleScss {
        sidePanelInfoContainer: string;
        sidePanelInfoTitle: string;
        sidePanelInfoHeading: string;
        sidePanelInfoIcon: string;
    }
  }

declare const MapIntroductionModuleScssModule: MapIntroductionModuleScssNamespace.IMapIntroductionModuleScss & {
    /** WARNING: Only available when `css-loader` is used without `style-loader` or `mini-css-extract-plugin` */
    locals: MapIntroductionModuleScssNamespace.IMapIntroductionModuleScss;
  };

export = MapIntroductionModuleScssModule;
