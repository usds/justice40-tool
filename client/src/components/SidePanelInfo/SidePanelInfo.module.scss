@use '../../styles/design-system.scss' as *;

.sidePanelInfoContainer {
  display: flex;
  flex-direction: column;
  @include u-padding-right(4);
  @include u-padding-left(4);
  @include u-padding-bottom(4);

  .sidePanelInfoTitle {
    @include u-padding-top(2);
    font-size: large;
    font-weight: 600;
  }
  
  .sidePanelInfoHeading {
    font-size: large;
    font-weight: 600;
  }
  
  .sidePanelInfoIcon {
    @include u-height(2);
  }
}
