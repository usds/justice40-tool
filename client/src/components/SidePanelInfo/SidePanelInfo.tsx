import React from 'react';
import {useIntl} from 'gatsby-plugin-intl';
import {useWindowSize} from 'react-use';

// @ts-ignore
import plusIcon from '/node_modules/uswds/dist/img/usa-icons/add.svg';
// @ts-ignore
import searchIcon from '/node_modules/uswds/dist/img/usa-icons/search.svg';
// @ts-ignore
import locateIcon from '/node_modules/uswds/dist/img/usa-icons/my_location.svg';
// @ts-ignore
import peopleIcon from '/node_modules/uswds/dist/img/usa-icons/people.svg';
// @ts-ignore
import fileUpIcon from '/node_modules/uswds/dist/img/usa-icons/file_upload.svg';
// @ts-ignore
import checkIcon from '/node_modules/uswds/dist/img/usa-icons/check.svg';
// @ts-ignore
import puzzleIcon from '../../images/sidePanelIcons/puzzle.svg';
// @ts-ignore
import bellCurveIcon from '../../images/sidePanelIcons/bell-curve.svg';
// @ts-ignore
import censusDotIcon from '../../images/sidePanelIcons/census-tract.svg';
// @ts-ignore
// import tribalDotIcon from '../../images/sidePanelIcons/tribal-tract.svg';
// @ts-ignore
import pieChartIcon from '../../images/sidePanelIcons/pie-chart.svg';
// @ts-ignore
// import handPointIcon from '../../images/sidePanelIcons/mouse-hand-point.svg';

import * as styles from './SidePanelInfo.module.scss';
import * as constants from '../../data/constants';
import * as EXPLORE_COPY from '../../data/copy/explore';

const SidePanelInfo = () => {
  const intl = useIntl();
  const {width: windowWidth} = useWindowSize();

  return (
    <aside className={styles.sidePanelInfoContainer}>

      {/* Heading 1 */}
      <header tabIndex={0} className={styles.sidePanelInfoTitle}>
        {intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_INITIAL_STATE.HEADING1)}
      </header>

      {/* Paragraph 1 */}
      <p tabIndex={0}>
        {intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_INITIAL_STATE.PARA1_PART1)}
        {
          windowWidth > constants.USWDS_BREAKPOINTS.MOBILE_LG &&
          <img tabIndex={0} className={styles.sidePanelInfoIcon} src={plusIcon}
            alt={intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_INIT_STATE_ICON_ALT_TEXT.PLUS)}
          />
        }
        {intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_INITIAL_STATE.PARA1_PART2)}
        {
          windowWidth > constants.USWDS_BREAKPOINTS.MOBILE_LG &&
          <img tabIndex={0} className={styles.sidePanelInfoIcon} src={searchIcon}
            alt={intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_INIT_STATE_ICON_ALT_TEXT.SEARCH)}
          />
        }
        {intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_INITIAL_STATE.PARA1_PART3)}
        {
          windowWidth > constants.USWDS_BREAKPOINTS.MOBILE_LG &&
          <img tabIndex={0} className={styles.sidePanelInfoIcon} src={locateIcon}
            alt={intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_INIT_STATE_ICON_ALT_TEXT.LOCATE)}
          />
        }
        {intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_INITIAL_STATE.PARA1_PART4)}
        {/* {
          windowWidth > constants.USWDS_BREAKPOINTS.MOBILE_LG &&
          <img tabIndex={0} className={styles.sidePanelInfoIcon} src={handPointIcon}
            alt={intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_INIT_STATE_ICON_ALT_TEXT.MOUSE)}
          />
        } */}
        {intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_INITIAL_STATE.PARA1_PART5)}
      </p>

      {/* Heading 2 */}
      <p tabIndex={0} className={styles.sidePanelInfoHeading}>
        {intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_INITIAL_STATE.HEADING2)}
      </p>

      {/* Paragraph 2 */}
      <p tabIndex={0}>
        {intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_INITIAL_STATE.PARA2_PART1)}
        {
          windowWidth > constants.USWDS_BREAKPOINTS.MOBILE_LG &&
          <img tabIndex={0} className={styles.sidePanelInfoIcon} src={puzzleIcon}
            alt={intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_INIT_STATE_ICON_ALT_TEXT.TRACT)}
          />
        }
        {intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_INITIAL_STATE.PARA2_PART2)}
        {
          windowWidth > constants.USWDS_BREAKPOINTS.MOBILE_LG &&
          <img tabIndex={0} className={styles.sidePanelInfoIcon} src={peopleIcon}
            alt={intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_INIT_STATE_ICON_ALT_TEXT.PEOPLE)}
          />
        }
        {intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_INITIAL_STATE.PARA2_PART3)}
      </p>

      {/* Paragraph 3 */}
      {
        windowWidth > constants.USWDS_BREAKPOINTS.MOBILE_LG &&
        <p tabIndex={0}>
          {intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_INITIAL_STATE.PARA3_PART1)}
          <img tabIndex={0} className={styles.sidePanelInfoIcon} src={censusDotIcon}
            alt={intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_INIT_STATE_ICON_ALT_TEXT.DAC_CIRCLE)}
          />
          {intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_INITIAL_STATE.PARA3_PART2)}
        </p>
      }

      {/* Paragraph 4 */}
      {
        windowWidth > constants.USWDS_BREAKPOINTS.MOBILE_LG &&
        <p tabIndex={0}>
          {intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_INITIAL_STATE.PARA4_PART1)}
          <img tabIndex={0} className={styles.sidePanelInfoIcon} src={bellCurveIcon}
            alt={intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_INIT_STATE_ICON_ALT_TEXT.BELL_CURVE)}
          />
          {intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_INITIAL_STATE.PARA4_PART2)}
        </p>
      }

      {/* Paragraph 5 */}
      {
        windowWidth > constants.USWDS_BREAKPOINTS.MOBILE_LG &&
       <p tabIndex={0}>
         {intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_INITIAL_STATE.PARA5_PART1)}
         <img tabIndex={0} className={styles.sidePanelInfoIcon} src={fileUpIcon}
           alt={intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_INIT_STATE_ICON_ALT_TEXT.FILE_UP)}
         />
         {intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_INITIAL_STATE.PARA5_PART2)}
         <img tabIndex={0} className={styles.sidePanelInfoIcon} src={pieChartIcon}
           alt={intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_INIT_STATE_ICON_ALT_TEXT.PIE_CHART)}
         />
         {intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_INITIAL_STATE.PARA5_PART3)}
         <img tabIndex={0} className={styles.sidePanelInfoIcon} src={checkIcon}
           alt={intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_INIT_STATE_ICON_ALT_TEXT.CHECK)}
         />
         {intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_INITIAL_STATE.PARA5_PART4)}
       </p>
      }

      {/* Paragraph 6 */}
      {
        windowWidth > constants.USWDS_BREAKPOINTS.MOBILE_LG &&
        <p tabIndex={0}>
          {intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_INITIAL_STATE.PARA6_PART1)}
          <img tabIndex={0} className={styles.sidePanelInfoIcon} src={censusDotIcon}
            alt={intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_INIT_STATE_ICON_ALT_TEXT.DAC_CIRCLE)}
          />
          {intl.formatMessage(EXPLORE_COPY.SIDE_PANEL_INITIAL_STATE.PARA6_PART2)}
        </p>
      }
    </aside>
  );
};

export default SidePanelInfo;
