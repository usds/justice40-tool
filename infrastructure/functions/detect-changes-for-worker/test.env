REGION=us-east-1
STAGE=sit

# ESC Cluster name that will from the containers
ECS_CLUSTER=j40-sit-justice40-data-harvester-ECSCluster-ktXGGU9zjwkb

# VPC Private Subnet that has a NAT GAteway
VPC_SUBNET_ID=subnet-07e68cb57322f7b1f

# Names of the container and task names.
GDAL_TASK_DEFINITION=sit-justice40-data-harvester-gdal
GDAL_CONTAINER_DEFINITION=sit-justice40-data-harvester-osgeo-gdal
TIPPECANOE_TASK_DEFINITION=sit-justice40-data-harvester-tippecanoe
TIPPECANOE_CONTAINER_DEFINITION=sit-justice40-data-harvester-mgiddens-tippecanoe

# AWS Credentials
AWS_ACCESS_KEY_ID=
AWS_SECRET_ACCESS_KEY=
AWS_DEFAULT_REGION=us-east-1