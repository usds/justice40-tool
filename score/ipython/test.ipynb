{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "1a4c0c68",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "70b3a793",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('data/fips_states.csv')  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "c514aad8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fips</th>\n",
       "      <th>state_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>Alabama</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>Alaska</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>Arizona</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5</td>\n",
       "      <td>Arkansas</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>California</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   fips  state_name\n",
       "0     1    Alabama \n",
       "1     2     Alaska \n",
       "2     4    Arizona \n",
       "3     5   Arkansas \n",
       "4     6  California"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b9ee44d9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
