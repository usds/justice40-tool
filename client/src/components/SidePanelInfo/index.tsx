import SidePanelInfo from './SidePanelInfo';

export default SidePanelInfo;
