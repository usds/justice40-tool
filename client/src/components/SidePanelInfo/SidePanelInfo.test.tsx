import * as React from 'react';
import {render} from '@testing-library/react';
import SidePanelInfo from './SidePanelInfo';
import {LocalizedComponent} from '../../test/testHelpers';

describe('rendering of the component', () => {
  const {asFragment} = render(
      <LocalizedComponent>
        <SidePanelInfo />
      </LocalizedComponent>,
  );

  it('expects the render to match snapshot', () => {
    expect(asFragment()).toMatchSnapshot();
  });
});
