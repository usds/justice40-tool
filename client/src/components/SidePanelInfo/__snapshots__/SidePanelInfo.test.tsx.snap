// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering of the component expects the render to match snapshot 1`] = `
<DocumentFragment>
  <aside>
    <header
      tabindex="0"
    >
      Things to know
    </header>
    <p
      tabindex="0"
    >
      
      This tool identifies communities that are marginalized, underserved, and overburdened by pollution. These communities are located in census tracts that are at or above the thresholds in one or more of eight categories of criteria.
    
    </p>
    <p
      tabindex="0"
    >
      
      Zoom in or search and select to see data about any census tract of interest
    
    </p>
    <img
      alt="
      An icon that has depicts pieces of a block selected mimicing the census block census tracts
    "
      src="test-file-stub"
      tabindex="0"
    />
    <p
      tabindex="0"
    >
      
      The tool uses census tracts that represent about 4,000 people, which is the smallest unit of geography for which consistent data can be displayed on the map.
    
    </p>
    <img
      alt="
      An icon that a bell curve or gaussian distribution
    "
      src="test-file-stub"
      tabindex="0"
    />
    <p
      tabindex="0"
    >
      
      The tool ranks each census tract using percentiles that show how much burden each tract experiences relative to all other tracts, for each criteria.
    
    </p>
    <img
      alt="
      An icon that depicts a part of pie chart being removed
    "
      src="test-file-stub"
      tabindex="0"
    />
    <p
      tabindex="0"
    >
      
      Percentages are used for certain variables, i.e. those relating to high school achievement rate and to the share of individuals not currently enrolled in higher education.
    
    </p>
    <img
      alt="
      An icon that has an up arrow and a down arrow
    "
      src="test-file-stub"
      tabindex="0"
    />
    <p
      tabindex="0"
    >
      
    Thresholds for each category determine if a tract should be identified as disadvantaged because it has exceeded a certain value for the relevant indicators.
  
    </p>
  </aside>
</DocumentFragment>
`;
